manifests/git_repo.pp

```diff
@@ -0,0 +1,22 @@
+# Setup and configure a git repo
+define katello_devel::git_repo(
+  $source,
+  $github_username = undef,
+) {
+
+  validate_string($source)
+
+  $sources = {origin => "https://github.com/${source}.git"}
```

Comment:
I'm in a similar position, but I use my username instead of origin. ```hub fork``` sets it up in the same way.